### PR TITLE
Merge master HEAD into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Resolved conflicts:
```
	src/java.base/share/classes/java/lang/Thread.java
```

There is a build issue since https://openj9-jenkins.osuosl.org/job/Build_JDKnext_aarch64_mac_OpenJDK/78/ (Apr 3rd)
* https://github.com/eclipse-openj9/openj9/issues/17121

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by keithc-ca** [comment]
This should target the openj9-staging branch. Please adjust accordingly.

File: src/java.base/share/classes/java/lang/Thread.java
Comment (keithc-ca) [inline src/java.base/share/classes/java/lang/Thread.java:1877]: For consistency, this should use `NO_REF` instead of `0`.
Please remove the method `isAlive0()` that is no longer used.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/lang/Thread.java
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java]: Moved `isAlive0()` implementation into `alive()`.

File: src/java.base/share/classes/java/lang/Thread.java
Comment (keithc-ca) [inline src/java.base/share/classes/java/lang/Thread.java:1878]: There's no longer any value in using `synrchronized` here now that `eetop` is `volatile`.

Comment (JasonFengJ9) [reply src/java.base/share/classes/java/lang/Thread.java]: Removed synchronization on `interruptLock` for `alive()`.

**Comment by keithc-ca** [comment]
Jenkins compile  zlinux jdknext depends https://github.com/eclipse-openj9/openj9/pull/17126

**Review by keithc-ca — APPROVED** [review]
I will wait until https://github.com/eclipse-openj9/openj9/pull/17126 is merged before merging this.

**Comment by pshipton** [comment]
Merging this introduces [crashes in OpenJDK testing](https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/581#issuecomment-1505475824). This doesn't need to block merging, but we'll have to figure it out at some point and it might be easier to do it now than a few months later.

**Comment by keithc-ca** [comment]
I reproduced the crashes referenced above; see https://github.com/eclipse-openj9/openj9/issues/17189.

**Pull request merged**
